pandas/core/methods/to_dict.py

```diff
@@ -106,15 +106,15 @@ def to_dict(
         return into_c((k, v.to_dict(into)) for k, v in df.items())
 
     elif orient == "list":
-        object_dtype_indices_as_set = set(box_native_indices)
+        object_dtype_indices_as_set: set[int] = set(box_native_indices)
         return into_c(
             (
-                k,
-                list(map(maybe_box_native, v.tolist()))
+                col,
+                list(map(maybe_box_native, df[col].to_numpy().tolist()))
                 if i in object_dtype_indices_as_set
-                else v.tolist(),
+                else df[col].to_numpy().tolist(),
             )
-            for i, (k, v) in enumerate(df.items())
+            for i, col in enumerate(df.columns)
```

Comment:
Could you add a test where `to_dict("list")` is called on a DataFrame with duplicate column labels?

Reply:
thanks for the catch, didin think about it